kivy/uix/behaviors/compoundselection.py

```diff
@@ -308,7 +308,7 @@ def select_with_touch(self, node, touch=None):
             return False
 
         if (node in self.selected_nodes and (not range_select)):  # selected
-            if multiselect:
+            if ((not multi) or multiselect):
                 self.deselect_node(node)
             else:
                 self.clear_selection()
```

Comment:
@dessant No actually what happens is deleting line 315 works for multiselect=False but it causes another problem that when self.multiselect=True and touch_multiselect=False(suppose) then if one doesn't presses the shift key then range_select=False => and when  one clicked on one of the selected nodes(out of many) then the first if condition evaluates to True . The multiselect in line 291 is False since he/she hasn't pressed either ctrl btn => it will clear all selections in line 314 and even won't select the node which was clicked on but the behaviour should be that the node which is clicked should remain selected and others should get deselected. Thats why that line has to be there.